target/linux/ramips/image/mt7621.mk

```diff
@@ -466,6 +466,20 @@ define Device/elecom_wrc-1900gst
 endef
 TARGET_DEVICES += elecom_wrc-1900gst
 
+define Device/elecom_wrc-2533ghbk-i
+  $(Device/dsa-migration)
+  $(Device/uimage-lzma-loader)
+  DEVICE_VENDOR := ELECOM
+  DEVICE_MODEL := WRC-2533GHBK-I
+  IMAGE_SIZE := 9856k
+  IMAGES += factory.bin
+  IMAGE/factory.bin := $$(sysupgrade_bin) | check-size | \
+	elx-header 0107002d 8844A2D168B45A2D | \
+	elecom-product-header WRC-2533GHBK-I $$$$@
```

Comment:
Not sure whether I like this concept if it requires handing over $$$$@ here ...

Reply:
The following change can be done (and tested compiling for WRC-2533GHBK-I and WRC-300GHBK2-I):

```
diff --git a/include/image-commands.mk b/include/image-commands.mk
index 0ebb0989b9..f167bf8886 100644
--- a/include/image-commands.mk
+++ b/include/image-commands.mk
@@ -157,7 +157,7 @@ endef

 define Build/elecom-product-header
        $(eval product=$(word 1,$(1)))
-       $(eval fw=$(word 2,$(1)))
+       $(eval fw=$(if $(word 2,$(1)),$(word 2,$(1)),$@))

        ( \
                echo -n -e "ELECOM\x00\x00$(product)" | dd bs=40 count=1 conv=sync; \
diff --git a/target/linux/ramips/image/mt7621.mk b/target/linux/ramips/image/mt7621.mk
index 5f0e2e4219..de684e8407 100644
--- a/target/linux/ramips/image/mt7621.mk
+++ b/target/linux/ramips/image/mt7621.mk
@@ -475,7 +475,7 @@ define Device/elecom_wrc-2533ghbk-i
   IMAGES += factory.bin
   IMAGE/factory.bin := $$(sysupgrade_bin) | check-size | \
        elx-header 0107002d 8844A2D168B45A2D | \
-       elecom-product-header WRC-2533GHBK-I $$$$@
+       elecom-product-header WRC-2533GHBK-I
   DEVICE_PACKAGES := kmod-mt7615e kmod-mt7615-firmware
 endef
 TARGET_DEVICES += elecom_wrc-2533ghbk-i
```